java/compaction/compaction-job-execution/src/main/java/sleeper/compaction/job/execution/CompactionTask.java

```diff
@@ -62,7 +74,7 @@ public void runAt(Instant startTime) throws InterruptedException, IOException {
                     LOGGER.info("Terminating compaction task as it was idle for {}, exceeding maximum of {}",
                             LoggedDuration.withFullOutput(runTime),
                             LoggedDuration.withFullOutput(maxIdleTime));
-                    return;
+                    break;
```

Comment:
It might read better if we split this method up so that this can still be a return statement? This method has got a bit big.

I also notice that in this case, the current time gets checked again from the time supplier immediately, to compute the total run time and task statistics.

It seems like the only case where we need to update the current time at the end is when we reach the maximum consecutive failures?